dbms/src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -704,6 +705,38 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
     return {};
 }
 
+BlockIO InterpreterCreateQuery::createDictionary(ASTCreateQuery & create)
+{
+    String dictionary_name = create.table;
+
+    String database_name = !create.database.empty() ? create.database : context.getCurrentDatabase();
+
+    auto guard = context.getDDLGuard(database_name, dictionary_name);
+    DatabasePtr database = context.getDatabase(database_name);
+
+    if (database->isDictionaryExist(context, dictionary_name))
+    {
+        if (create.if_not_exists)
```

Comment:
TODO in comment if we are going to check Dictionary structure in future.

Reply:
Check that it's the same dictionary?

Reply:
If we have dictionary with the same name but different structure it leads to not idempotent behavior. It's OK for local DBMS not to throw but it's error prone for distributed ones. We probably need a setting to throw on ```CREATE TABLE|DICTIONARY IF NOT EXISTS``` with different structure. Something like ```idempotent_not_exists```

Reply:
Good idea! More important for tables, than for dictionaries.